From 9218d6cb2ce3cc9bfd359e5ce4c2cc6780e89598 Mon Sep 17 00:00:00 2001
From: Steve Sakoman <steve@sakoman.com>
Date: Wed, 5 Oct 2011 12:34:39 -0700
Subject: [PATCH 1/2] net: eth.c: Restore previous behavior for
 eth_getenv_enetaddr_by_index failure

Recently added error check causes "Warning: failed to set MAC address" on
systems that set the MAC via eeprom

Just a temporary workaround till correct fix is determined
---
 net/eth.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/net/eth.c b/net/eth.c
index 4280d6d..7f0a94f 100644
--- a/net/eth.c
+++ b/net/eth.c
@@ -190,8 +190,7 @@ int eth_write_hwaddr(struct eth_device *dev, const char *base_name,
 	unsigned char env_enetaddr[6];
 	int ret = 0;
 
-	if (!eth_getenv_enetaddr_by_index(base_name, eth_number, env_enetaddr))
-		return -1;
+	eth_getenv_enetaddr_by_index(base_name, eth_number, env_enetaddr);
 
 	if (memcmp(env_enetaddr, "\0\0\0\0\0\0", 6)) {
 		if (memcmp(dev->enetaddr, "\0\0\0\0\0\0", 6) &&
-- 
1.7.6.4

